cpp/tests/utilities_tests/column_utilities_tests.cpp

```diff
@@ -173,27 +175,21 @@ TEST_F(ColumnUtilitiesStringsTest, StringsToHostAllNulls)
 TYPED_TEST(ColumnUtilitiesTestNumeric, PrintColumnNumeric) {
   const char* delimiter = ",";
 
-  cudf::test::fixed_width_column_wrapper<TypeParam> cudf_col({1, 2, 3, 4, 5});
-  std::vector<TypeParam>                            std_col({1, 2, 3, 4, 5});
+  cudf::test::fixed_width_column_wrapper<TypeParam, int32_t> cudf_col({1, 2, 3, 4, 5});
+  auto std_col = cudf::test::make_type_param_vector<TypeParam>({1, 2, 3, 4, 5});
 
-  std::ostringstream tmp;
-
-  int index = 0;
-  for (auto x : std_col) {
-    tmp << ((index == 0) ? "" : delimiter);
-    tmp << std::to_string(x);
-    ++index;
-  }
+  auto to_string = [] (auto e) { return std::to_string(e); };
+  auto const expected = boost::join(std_col | boost::adaptors::transformed(to_string), delimiter);
```

Comment:
libcudf is not dependent on boost, please remove this. (Yes, arrow is dependent on boost, but we have successfully avoided a direct libcudf->boost dependency up to now.)

Reply:
We do use `boost::filesystem` in `jit/cache.cpp` to manage the JIT cache directory: https://github.com/rapidsai/cudf/blob/fc4de2c76b8bab91b9274f14853618302fb02ede/cpp/src/jit/cache.cpp#L46-L51

Reply:
Also we link against `boost` libraries (`cpp/CMakeLists.txt`):
```
target_link_libraries(cudf NVCategory NVStrings rmm ${ARROW_CUDA_LIB_LINK} ${ARROW_LIB} nvrtc cudart cuda $\
   {ZLIB_LIBRARIES} ${**Boost_LIBRARIES**})
```

We certainly have a strong dependency in the `cpp/CMakeLists.txt`: 
```
message(FATAL_ERROR "Boost not found, please check your settings.")
``` 
if `boost` not found.

Reply:
Hmmm, didn't realize that. The boost::filesystem dependency will be removed when we can move up to C++17. I don't think this use of boost really improves the code. In fact, since I've never used those features of boost, I don't know what the new code does. And since it's not `std::`, it's not trivial to look up.  I would stick to std:: here.

Reply:
As I posted on Slack, unfortunately, this code:
```cpp
  std::stringstream tmp;

  std::copy(
    std::begin(std_col),
    std::prev(std::end(std_col)),
    std::ostream_iterator<TypeParam>(tmp, delimiter));

  tmp << std::to_string(std_col.back());
```
Doesn't work for all types. It fails for `float`, `double` and `signed char`. You would first need to call `std::transform` on a temporary `vector` or use a transform iterator.
I have changed it back to the old code.

Reply:
What's wrong with a transform iterator?

Reply:
Nothing, I mentioned it in the Slack thread. I switched it to the `std::copy` + a `thrust::transform_iterator`

Reply:
Nothing, I mentioned it in the Slack thread. I switched it to the `std::copy` + a `thrust::transform_iterator`